shell/spy.go

```diff
@@ -2,18 +2,37 @@ package shell
 
 import (
 	"bytes"
-	"log"
+	"fmt"
+	"io"
 	"os/exec"
+
+	"bitbucket.org/creachadair/shell"
 )
 
-// Execute accepts a command argument and returns the result of the executed command
-func Execute(script string) string {
-	cmd := exec.Command(script)
+// ParseCommand accepts a string and returns the command name and args
+func ParseCommand(cmd string) (string, []string) {
+	stringResult, _ := shell.Split(cmd)
```

Comment:
We need to check the second value here, don't we? Because the command line could have unbalanced quotes.

(Use of the blank identifier to ignore some value is almost always a mistake, and when it's not, it's important to comment to say _why_ we assume it's all right to ignore. Our assumptions could be wrong, or become outdated due to later code changes, which is why we need to surface them.)

Reply:
Hmm...my take on this is we don't really want to _filter_ the command that comes in since we're looking to record the input and output of the actual os shell, right? We really just want to mimic the shell behaviour to be able to spy on the activities. Or am I missing something fundamental here?